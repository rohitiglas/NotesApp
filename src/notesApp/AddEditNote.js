import React, {useEffect, useState, memo} from 'react';
import {TextInput, Text, View, TouchableOpacity, StatusBar} from 'react-native';
import styles from '../login/LoginStyle';
const Form = ({
  showModal,
  toggleModal,
  note,
  addNote,
  updateNote,
  inputs,
  setInputs,
  closeModal
}) => {
  const [title, setTitle] = useState('');
  const [description, setDescription] = useState('');
  useEffect(() => {
    if (note) {
      setTitle(note.title);
      setDescription(note.body);
    }
  }, [note, setInputs]);
  const handleSubmit = () => {
    console.log("HandleSubmit",note);
    if (!note.id) {
      addNote(title, description);
      return;
    }
    updateNote(note.id, title, description);
  };
  return (
    <View testID='add-edit-page' style={styles.mainContainer}>
      <View style={styles.loginContainer}>
        <StatusBar barStyle="light-content" />
        <TextInput
          testID={'input-title'}
          value={title}
          onChangeText={text => setTitle(text)}
          placeholder={'Title'}
          multiline={true}
          numberOfLines={2}
          maxLength={100}
          style={{
            height: 80,
            width: '80%',
            fontSize: 20,
            backgroundColor: 'white',
            marginBottom: 20,
            borderRadius: 20,
            padding: 10,
            color: '#2c3e50',
            paddingTop: 20,
          }}
        />
        <TextInput
          testID={'input-description'}
          value={description}
          onChangeText={text => setDescription(text)}
          placeholder={'Body'}
          multiline={true}
          numberOfLines={10}
          maxLength={1000}
          style={{
            height: 200,
            width: '80%',
            fontSize: 20,
            backgroundColor: 'white',
            marginBottom: 20,
            borderRadius: 20,
            padding: 10,
            paddingTop: 20,
            color: '#2c3e50',
            textAlignVertical: 'top',
          }}
        />
        <TouchableOpacity
          testID={'submit-button'}
          style={styles.buttonContainer}
          onPress={handleSubmit}>
          <Text style={styles.buttonText}>
            {note.id ? 'Update Note' : 'Add Note'}
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          testID={'close-button'}
          style={{position:'absolute',top:40,right:20,marginTop:20,width:50,height:50,borderRadius:25,backgroundColor:'#000',alignItems:'center',justifyContent: 'center'}}
          onPress={closeModal}>
          <Text style={styles.buttonText}>
            X
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};
export default memo(Form);
