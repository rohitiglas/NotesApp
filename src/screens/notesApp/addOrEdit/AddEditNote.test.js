import 'react-native';
import React from 'react';
import {
  render,
  cleanup,
  fireEvent,
  flushMicrotasksQueue,
} from '@testing-library/react-native';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import AddEditNote from './AddEditNote';

let mock = new MockAdapter(axios);

describe('Add Note Page', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = render(
      <AddEditNote
        showModal={jest.fn()}
        note={{}}
        closeModal={jest.fn()}
        toggleModal={jest.fn()}
        addNote={jest.fn()}
        updateNote={jest.fn()}
      />,
    );
  });
  afterEach(() => {
    cleanup();
    wrapper = null;
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call Add Note api when you click add Button', async function () {
    const {getByTestId} = wrapper;
    const title = getByTestId('input-title');
    const description = getByTestId('input-description');
    const submitButton = getByTestId('submit-button');
    expect(getByTestId('button-text').children[0]).toBe('Add Note');
    fireEvent.changeText(title, 'My Notes');
    fireEvent.changeText(description, 'This is my First Note');
    fireEvent.press(submitButton);
  });
  it('should close Modal when you click Close Button', async function () {
    const {getByTestId} = wrapper;
    const closeButton = getByTestId('close-button');
    fireEvent.press(closeButton);
  });

  it('should call Update Note api when you click Update Button', async function () {
    wrapper = render(
      <AddEditNote
        showModal={jest.fn()}
        note={{title: 'Rohit', body: 'Bansal', id: '10'}}
        closeModal={jest.fn()}
        toggleModal={jest.fn()}
        addNote={jest.fn()}
        updateNote={jest.fn()}
      />,
    );
    const {getByTestId} = wrapper;
    const title = getByTestId('input-title');
    const description = getByTestId('input-description');
    const submitButton = getByTestId('submit-button');
    expect(getByTestId('button-text').children[0]).toBe('Update Note');
    expect(title.props.value).toBe('Rohit');
    expect(description.props.value).toBe('Bansal');
    fireEvent.press(submitButton);
  });
});
