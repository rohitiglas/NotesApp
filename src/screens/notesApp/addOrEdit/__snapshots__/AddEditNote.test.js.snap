// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Note Page renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2c3e50",
      "flex": 1,
    }
  }
  testID="add-edit-page"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      maxLength={100}
      multiline={true}
      numberOfLines={2}
      onChangeText={[Function]}
      placeholder="Title"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 20,
          "color": "#2c3e50",
          "fontSize": 20,
          "height": 80,
          "marginBottom": 20,
          "padding": 10,
          "paddingTop": 20,
          "width": "80%",
        }
      }
      testID="input-title"
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      maxLength={1000}
      multiline={true}
      numberOfLines={10}
      onChangeText={[Function]}
      placeholder="Body"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 20,
          "color": "#2c3e50",
          "fontSize": 20,
          "height": 200,
          "marginBottom": 20,
          "padding": 10,
          "paddingTop": 20,
          "textAlignVertical": "top",
          "width": "80%",
        }
      }
      testID="input-description"
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#2980b6",
          "borderRadius": 20,
          "opacity": 1,
          "paddingVertical": 15,
          "width": "80%",
        }
      }
      testID="submit-button"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontWeight": "700",
            "textAlign": "center",
          }
        }
        testID="button-text"
      >
        Add Note
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#000",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "position": "absolute",
          "right": 20,
          "top": 40,
          "width": 50,
        }
      }
      testID="close-button"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontWeight": "700",
            "textAlign": "center",
          }
        }
      >
        X
      </Text>
    </View>
  </View>
</View>
`;
